src/Formik.tsx

```diff
@@ -393,9 +394,12 @@ export class Formik<ExtraProps = {}, Values = object> extends React.Component<
   submitForm = () => {
     // Recursively set all values to `true`.
     this.setState(prevState => ({
-      touched: setNestedObjectValues<FormikTouched<Values>>(
-        prevState.values,
-        true
+      touched: deepmerge(
+        setNestedObjectValues<FormikTouched<Values>>(prevState.values, true),
+        Object.keys(this.fields).reduce(
+          (prev, curr) => ({ ...prev, [curr]: true }),
```

Comment:
this.fields could contain deep paths like `social.facebook`, does this address this?

Reply:
more explicitly, it could contain something like: 

```
{
  name: ...,
  'social.facebook': ...
}
```

Reply:
nope, haven't considered this use case. Will test that, no prob.